xblock/test/test_fields_api.py

```diff
@@ -163,12 +163,26 @@ def test_delete_with_save_writes(self):
         assert_false(self.field_data.has(self.block, 'field'))
         assert_true(self.is_default())
 
-    def test_set_after_get_always_saves(self):
+    def test_set_after_get_always_force_saves(self):
         with patch.object(self.field_data, 'set_many') as patched_set_many:
             self.set(self.get())
-            self.block.save()
 
-            patched_set_many.assert_called_with(self.block, {'field': self.get()})
+            self.block.force_save_fields(['field'])
+
+            patched_set_many.assert_called_with(
+                self.block, {'field': self.get()}
+            )
+
+    def test_set_after_get_doesnt_save(self):
+        self.set(self.get())
+        # pylint: disable=protected-access
+        fields_to_save = self.block._get_fields_to_save()
+        assert_equals(len(fields_to_save), 0)
+
+        self.set(self.new_value)
+        # pylint: disable=protected-access
+        fields_to_save = self.block._get_fields_to_save()
+        assert_equals(len(fields_to_save), 1)
```

Comment:
I think this test would be better written to test that `patched_set_many` is or isn't called after each call to `save`.